eclair-core/src/main/scala/fr/acinq/eclair/channel/publish/TxPublisher.scala

```diff
@@ -86,7 +85,7 @@ object TxPublisher {
     def apply(txInfo: TransactionWithInputInfo, fee: Satoshi, parentTx_opt: Option[ByteVector32]): PublishFinalTx = PublishFinalTx(txInfo.tx, txInfo.input.outPoint, txInfo.desc, fee, parentTx_opt)
   }
   /** Publish an unsigned transaction that can be RBF-ed. */
-  case class PublishReplaceableTx(txInfo: ReplaceableTransactionWithInputInfo, commitments: Commitments) extends PublishTx {
+  case class PublishReplaceableTx(txInfo: ReplaceableTransactionWithInputInfo, commitments: Commitments, deadline: Long) extends PublishTx {
```

Comment:
Good idea, I can define a `BlockHeight` type right now and use it for these commands.
In a follow-up PR, we can then migrate the rest of the codebase to use this wrapper type.
I'll rename this `confirmationTarget` instead of deadline.

> even if they are on time, we can still compete against their transaction before they confirm

It depends: for htlcs txs, if they are on time and get their transactions confirmed before this deadline/confirmationTarget, we don't have a competing transaction (because being on time means they confirm before the htlc timeout).

For commit txs though, they do have a competing transaction (local commit vs remote commit), so I agree that `deadline` can be a misleading name.

Reply:
Done in https://github.com/ACINQ/eclair/pull/2113/commits/b0667538d8dbf95c15b4c77c042787ba9930ea59

Reply:
This got me thinking about the relationship between `CltvExpiry` and `BlockHeight`, and now I'm wondering if we shouldn't have just renamed `CltvExpiry` to `BlockHeight`.